### ISSUE TITLE
1st column showing dynamic loading data

### ISSUE THREAD

**Issue by xtreemrage** [khan4019/tree-grid-directive#107]
When I loading data from a server or external json file, I only see the first column. This behavior is not expected.

I saw this issue: [36](https://github.com/khan4019/tree-grid-directive/issues/36), but this is not a solution but a work around, and works only if you know the columns header before hand. I don't know that, so is there a solution for this problem?

Here a [plunker](https://plnkr.co/IgRdktNG5Ymp2DYfaVwf?p=preview) (give it a moment to load)

**Comment by mizhon** [comment]
Yes, I got the same issue, the solution for myself is assign the basic format instead of the empty array (`vm.treedata=[]`). something like this: `vm.treeData = [{id: '', name: '', permission: '', sort: '', url: '', children: []}];` In this case, you could load the data noramlly.

**Comment by chamma** [comment]
I also ran into this issue and mizhons solution worked for me. However this still means, all possible fields must be known in advance. I suspect this is a rather frequent use case of the tree view and it took me a while to solve the issue, so it would be good to fix this. I'm new to angular, so unfortunately I cannot provide a fix myself.
